src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -359,4 +357,22 @@ private void doWithZkUtils(final ZkUtilsAction action) throws Exception {
             }
         }
     }
+
+    private Integer calculateKafkaPartitionCount(final EventTypeStatistics stat) {
+        if (null == stat) {
+            return settings.getDefaultTopicPartitionCount();
+        }
+        final int maxPartitionsDueParallelism = Math.max(stat.getReadParallelism(), stat.getWriteParallelism());
+        if (maxPartitionsDueParallelism >= settings.getMaxTopicPartitionCount()) {
+            return settings.getMaxTopicPartitionCount();
+        }
+        return Math.min(8, Math.max(
```

Comment:
nope, they are not redundant. I just do not want to do calculations if they are not needed. In case you provided `calculatePartitionsAccordingLoad` will be called even if it won't work at all. But I can change it anyway.
The problem here is that I forgot to replace `8` with `getMaxTopicPartitionCount` call